modules/social_features/social_core/tests/src/Kernel/ExternalIdFieldTypeTest.php

```diff
@@ -0,0 +1,325 @@
+    $this->installEntitySchema('test_external_owner_entity');
+    $this->installConfig(['field', 'node', 'user']);
+
+    // Create a node type.
+    $node_type = NodeType::create([
+      'type' => 'test',
+      'name' => 'Test',
+    ]);
+    $node_type->save();
+
+    // Create the field storage.
+    $field_storage_field_social_external_identifier = FieldStorageConfig::create([
+      'field_name' => 'field_social_external_identifier',
+      'entity_type' => 'node',
+      'type' => 'social_external_identifier',
+      'settings' => [
+        'target_types' => [
+          'test_external_owner_entity' => 'test_external_owner_entity',
+        ],
+      ],
+      'cardinality' => FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED,
+    ]);
+    $field_storage_field_social_external_identifier->save();
+    // Create the field.
+    $field_field_social_external_identifier = FieldConfig::create([
+      'field_storage' => $field_storage_field_social_external_identifier,
+      'bundle' => 'test',
+      'label' => 'External Identifier example',
+    ]);
+    $field_field_social_external_identifier->save();
+
+    // Create the field storage for field without target types defined.
+    $field_storage_field_sei_no_target_types = FieldStorageConfig::create([
+      // `sei` stands for social_external_identifier.
+      'field_name' => 'field_sei_no_target_types',
+      'entity_type' => 'node',
+      'type' => 'social_external_identifier',
+      'settings' => [
+        'target_types' => [],
+      ],
+      'cardinality' => FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED,
+    ]);
+    $field_storage_field_sei_no_target_types->save();
+    // Create the field.
+    $field_field_storage_field_sei_no_target_types = FieldConfig::create([
+      'field_storage' => $field_storage_field_sei_no_target_types,
+      'bundle' => 'test',
+      'label' => 'External Identifier example',
+    ]);
+    $field_field_storage_field_sei_no_target_types->save();
+  }
+
+  /**
+   * Tests that the social_external_identifier settings are correctly applied.
+   */
+  public function testFieldSettings(): void {
+    // Load the field storage configuration and verify settings.
+    /** @var \Drupal\field\FieldStorageConfigInterface $field_storage */
+    $field_storage = FieldStorageConfig::loadByName('node', 'field_social_external_identifier');
+    $this->assertEquals(['test_external_owner_entity' => 'test_external_owner_entity'], $field_storage->getSettings()['target_types']);
+  }
+
+  /**
+   * Tests that the field can be added to a node and stores values correctly.
+   *
+   * TOC:
+   *  - Case 1: Create a node with the external identifier field.
+   * - Case 2: Create a node without the external identifier field.
+   * - Case 3: Create a node with the external identifier field with subfield
+   * values set as NULL.
+   * - Case 4: Create a node with the external identifier field with subfield
+   * values set as empty string.
+   */
```

Comment:
I think it's great that you've already split these into separate cases. Is there a reason that they're not actually separate tests? If a developer has to debug this then it's very difficult to know whether case 2 or case 3 has broken and thus where to start looking for errors. 

A data provider may help if the structure of the test is the same but the inputs are different: https://blog.martinhujer.cz/how-to-use-data-providers-in-phpunit/

Also don't be afraid to have some duplicated set-up in your tests if it helps a failure provide more information :) 

The same goes for your other test which described multiple cases.

Reply:
Thank you for the feedback, will split them 💪

Reply:
All split. I did not used data providers as all tests are different. I kept the TOC as we have 10 kernel tests methods now ;) Marking as resolved.